pynamodb/models.py

```diff
@@ -5,9 +5,28 @@
 import time
 import logging
 import warnings
+import sys
 from inspect import getmembers
-from typing import Any, Dict, Generic, Iterable, Iterator, List, Optional, Sequence, Mapping, Type, TypeVar, Text, \
-    Tuple, Union, cast
+from typing import Any
+from typing import Dict
+from typing import Generic
+from typing import Iterable
+from typing import Iterator
+from typing import List
+from typing import Mapping
+from typing import Optional
+from typing import Sequence
+from typing import Text
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
+from typing import Union
+from typing import cast
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
```

Comment:
I get `ModuleNotFoundError: No module named 'typing_extensions'` on python3.6 without explicitly installing `typing-extensions`. If we want to do this, I think we'd have to add it to `setup.py` as required

Reply:
i added it to install_requires below

Reply:
Somehow missed this, my bad!